### PR TITLE
add ext filter

### LINKED ISSUE #47: Add filters similar to katana
> @ehsandeep @dogancanbakir 
> 
> Please consider adding the below features as well for more flexible output and to control junk results
> 
> ```console
> FILTER:
>    -mr, -match-regex string[]       regex or list of regex to match on output url (cli, file)
>    -fr, -filter-regex string[]      regex or list of regex to filter on output url (cli, file)
>    -f, -field string                field to display in output (url,path,fqdn,rdn,rurl,qurl,qpath,file,ufile,key,value,kv,dir,udir)
>    -em, -extension-match string[]   match output for given extension (eg, -em php,html,js)
>    -ef, -extension-filter string[]  filter output for given extension (eg, -ef png,css)
> ``` 

 _Originally posted by @zy9ard3 in [#17](https://github.com/projectdiscovery/urlfinder/issues/17#issuecomment-2525564340)_

### PR DESCRIPTION
**Pull request by dogancanbakir**
closes https://github.com/projectdiscovery/urlfinder/issues/47

## Summary by CodeRabbit

* **New Features**
  * Extension-based filtering for URL enumeration with CLI flags to include or exclude file extensions; option to disable the default deny-list.

* **Chores**
  * Quieted spurious error reporting from resource/network cleanup by ignoring close/flush errors to reduce noisy runtime logs.

* **Tests**
  * Added unit tests covering the extension validator behavior, normalization, and edge cases.

### CODE REVIEW

File: internal/runner/options.go [hunk 1]
```diff
@@ -60,6 +61,10 @@ type Options struct {
 	Filter             goflags.StringSlice
 	matchRegexes       []*regexp.Regexp
 	filterRegexes      []*regexp.Regexp
+	ExtensionsMatch    goflags.StringSlice
+	ExtensionFilter    goflags.StringSlice
+	NoDefaultExtFilter bool
+	extensionValidator *extensions.Validator
```
[the remark this answers is not in the document]
Comment (Mzack9999) [reply internal/runner/options.go:67]: @coderabbitai fix the name inconsistency

**Comment by Mzack9999** [comment]
@coderabbitai resolve

**Review by Mzack9999 — APPROVED** [review]
